account_bank_match/models/account_bank_statement.py

```diff
@@ -1102,6 +1102,11 @@ def write(self, vals):
         # FIXME: Values are not saved anymore when removing this method
         return super(AccountBankStatementLine, self).write(vals)
 
+    @api.multi
+    def _get_iban_country_id(self):
+        self.ensure_one()
+        return self.env['res.country'].search(
```

Comment:
What about bank accounts that are not IBAN?

Reply:
@NL66278 I see, so then first check if `sanitized_acc_number` is a valid IBAN, if not just use the country id.

Reply:
In most cases country_id will not be filled in res_partner_bank. So this would make the condition for this rule false for nearly all accounts that are not valid Iban. Is that OK? If not you might consider returning the partner_invoice_id.country_id, which is the same field you are comparing, so then this rule would be true for invalid IBAN with unknown country in res_partner_bank

Reply:
>In most cases country_id will not be filled in res_partner_bank. So this would make the condition for this rule false for nearly all accounts that are not valid Iban. Is that OK?

Yes that should be ok, that means that they do not get those extra points.

Note:

I see that the bank accounts are created by the odoo_toolbox scripts that fetch them from a mysql database on spiekerog. I am a little bit cautious on checking there but there should be more info about the accounts; possibly the country as well.

My proposal is to leave this as it is and make sure that the `country_id` field is mandatory. I could run a script to fill the country for the records that have iban (11528 out of 21068 on a test db I have) and for the rest if the field `country_id` has a value then ok, else do not do anything. I think proper input of the data should be a responsibility of the end user and we can possibly help with some restrictions


1200wd_addons/odoo_toolbox/ari_payment_import/payment_import.py